messaging/broker.go

```diff
@@ -774,6 +791,33 @@ func (t *topic) Close() error {
 	return nil
 }
 
+// loadIndex reads the highest available index for a topic from disk.
+func (t *topic) loadIndex() error {
```

Comment:
This seems like it's going to take a long time if there are quite a bit of topic logs backed up.